src/components/Header/Share.tsx

```diff
@@ -37,32 +24,63 @@ const Share = () => {
       .catch((err: string) => publish({ stderr: err }));
   }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
-  const shareRef = useRef<HTMLInputElement>(null);
+  const [modalOpen, setModalOpen] = useState<boolean>(false);
+  const toggleModalOpen = () => setModalOpen(!modalOpen);
 
-  const focusFn = () => {
-    shareRef.current?.focus();
-    shareRef.current?.setSelectionRange(0, shareRef.current.value.length);
-  };
+  const shareRef = useRef<HTMLInputElement>(null);
 
-  const shareUrl = (share: string) => (share !== '' ? window.location.href.split('#')[0].concat(`#${share}`) : '');
+  const shareUrl = (share?: string) => (share !== '' ? window.location.href.split('#')[0].concat(`#${share}`) : '');
 
   const shareFn = () => {
     storeState({ input, lineage }).then((id) => {
+      toggleModalOpen();
       setShareId(id);
-      setTimeout(focusFn, 500);
-      navigator.clipboard.writeText(shareUrl(id));
       window.location.href = shareUrl(id);
     });
   };
 
+  const onFocus = () => {
+    shareRef.current?.focus();
+    shareRef.current?.setSelectionRange(0, shareRef.current.value.length);
+  };
+
+  const onCopy = () => {
+    onFocus();
+
+    const url = shareUrl(shareId);
+    if (url !== '') {
+      navigator.clipboard.writeText(url);
+    }
+  };
+
+  const styles = useStyles2(getStyles);
+
   return (
     <>
-      <button style={styles.btn} onClick={shareFn}>
-        Share
-      </button>
-      <input style={styles.input} ref={shareRef} onClick={focusFn} readOnly={true} value={shareUrl(shareId || '')} />
+      <div className={styles.iconContainer}>
```

Comment:
I tried to set the `className` directly to the `IconButton`, but it gets overwritten by its default styles.